ios/AppLovinMAXAdView.m

```diff
@@ -35,6 +35,18 @@ @interface AppLovinMAXAdView()<MAAdViewAdDelegate, MAAdRevenueDelegate>
 
 @implementation AppLovinMAXAdView
 
+static NSMutableDictionary<NSString *, MAAdView *> *adViewInstances; // Static Instance for Amazon
+
++ (MAAdView *)sharedWithAdUnitIdentifier:(NSString *)adUnitIdentifier
+{
+    return adViewInstances[adUnitIdentifier];
+}
+
++ (void)initialize
+{
+    adViewInstances = [NSMutableDictionary dictionary];
+}
+
```

Comment:
Always call `[super initialize]`, and that is usualy always declared first since it initializes the class.

```
@implementation AppLovinMAXAdView
static NSMutableDictionary<NSString *, MAAdView *> *adViewInstances;

+ (void)initialize
{
    [super initialize];
    adViewInstances = [NSMutableDictionary dictionary];
}

+ (nullable MAAdView *)sharedWithAdUnitIdentifier:(NSString *)adUnitIdentifier
{
    return adViewInstances[adUnitIdentifier];
}
```